src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -97,8 +111,26 @@ protected override Assembly Load(AssemblyName assemblyName)
         {
             lock (_guard)
             {
+                // First try to satisfy the load from the default context.
+                Assembly assembly;
+                try
+                {
+                    assembly = _defaultContext.LoadFromAssemblyName(assemblyName);
+                    if (assembly != null)
+                    {
+                        return assembly;
+                    }
+                }
+                catch
+                {
+                    // LoadFromAssemblyName indicates failure by throwing exceptions; we need to
+                    // catch them, ignore them, and try to load the assembly ourselves.
+                    // This may mask underlying issues where _defaultContext.LoadFromAssemblyName
+                    // should have succeeded, but there isn't anything we can do about that.
```

Comment:
So there are a bunch of different exceptions that might be thrown, all indicating the same situation that this assembly isn't already loaded?